Source/WebKit/NetworkProcess/Downloads/cocoa/DownloadCocoa.mm

```diff
@@ -148,6 +148,16 @@
     if (enableModernDownloadProgress()) {
         bool isUsingPlaceholder = useDownloadPlaceholder == WebKit::UseDownloadPlaceholder::Yes;
 
+        if (!m_downloadTask.get().currentRequest.URL) {
+            RELEASE_LOG_ERROR(Network, "Download::publishProgress: Invalid request URL");
+            return;
+        }
+
+        if (!url.isValid()) {
```

Comment:
Also, checking `url.isValid()` is not quite equivalent to checking that `(NSURL *)url` is null I think. I'd suggest converting the URL to a NSURL early and null checking that.